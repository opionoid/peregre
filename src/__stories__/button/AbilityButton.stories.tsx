import { Story, Meta } from '@storybook/react'
import { AbilityButton, IAbilityButtonProps } from 'src/components/actor/button/AbilityButton'
import MOCK_ABILITY_BUTTON_PROPS from 'src/__mocks__/components/actor/button/AbilityButton'

export default {
  title: 'components/actor/button/AbilityButton',
  component: AbilityButton,
} as Meta

const Template: Story<IAbilityButtonProps> = (args) => <AbilityButton {...args} />
export const BattleMode = Template.bind({})
BattleMode.args = {
  ...MOCK_ABILITY_BUTTON_PROPS
}

export const EditMode = Template.bind({})
EditMode.args = {
  ...MOCK_ABILITY_BUTTON_PROPS,
  ...{ isEditMode: true }
}