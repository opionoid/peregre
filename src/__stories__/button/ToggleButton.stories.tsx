import { Story, Meta } from '@storybook/react'
import {
  ToggleButton,
  IToggleButtonProps,
} from 'src/components/actor/button/ToggleButton'
import MOCK_TOGGLE_BUTTON_PROPS from 'src/__mocks__/components/actor/button/ToggleButton'

export default {
  title: 'components/actor/button/ToggleButton',
  component: ToggleButton,
} as Meta

const Template: Story<IToggleButtonProps> = (args) => <ToggleButton {...args} />
export const Default = Template.bind({})
Default.args = {
  ...MOCK_TOGGLE_BUTTON_PROPS,
}
