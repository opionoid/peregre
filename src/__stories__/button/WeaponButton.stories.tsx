import { Story, Meta } from '@storybook/react'
import {
  WeaponButton,
  IWeaponButtonProps,
} from 'src/components/actor/button/WeaponButton'
import MOCK_WEAPON_BUTTON_PROPS from 'src/__mocks__/components/actor/button/WeaponButton'

export default {
  title: 'components/actor/button/WeaponButton',
  component: WeaponButton,
} as Meta

const Template: Story<IWeaponButtonProps> = (args) => <WeaponButton {...args} />
export const Default = Template.bind({})
Default.args = {
  ...MOCK_WEAPON_BUTTON_PROPS,
}
