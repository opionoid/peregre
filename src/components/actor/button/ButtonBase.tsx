import React from 'react'
import styled, { css } from 'styled-components'
import { color, space } from 'src/assets/style'

export interface IButtonBaseProps {
  children?: React.ReactNode
  isEditMode?: boolean
  type?: 'button' | 'submit' | 'reset'
  onClick?: React.MouseEventHandler<HTMLButtonElement>
  accent?: boolean
  lighten?: boolean
  name?: string
  value?: any
  disabled?: boolean
}

export const ButtonBase: React.FC<IButtonBaseProps> = ({
  children,
  isEditMode = false,
  type = 'button',
  onClick,
  accent = false,
  lighten = false,
  name,
  value,
  disabled = false
}) => {
  return (
    <ButtonBaseWrapper
      type={type}
      onClick={onClick}
      isEditMode={isEditMode}
      accent={accent}
      lighten={lighten}
      name={name}
      value={value}
      aria-disabled={disabled}
    >
      {children}
    </ButtonBaseWrapper>
  )
}

const ButtonBaseWrapper = styled.button<IButtonBaseProps>`
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: ${color.backgroundHighContrast};
  color: ${color.fontInHighContrast};
  padding: ${space.xs};
  border: 0;
  border-radius: inherit;
  transition: all ease-out 0.3s;
  font-size: 1.125rem;
  box-shadow: 2px 2px 6px ${color.backgroundLowContrastShadowDarken},
    -2px -2px 6px ${color.backgroundLowContrastShadowLighten};
  &:active,
  &[aria-pressed='true'] {
    box-shadow: inset 2px 2px 6px ${color.backgroundLowContrastShadowDarken},
      inset -2px -2px 6px ${color.backgroundLowContrastShadowLighten};
    transform: scale3d(0.98, 0.98, 0.99);
  }
  &[aria-disabled='true'] {
    cursor: not-allowed;
    background-color: ${color.backgroundLowContrast};
  }

  ${(props) =>
    props.isEditMode &&
    css`
      border-radius: 20px;
    `}

  ${(props) =>
    props.accent &&
    css`
      color: ${color.fontInHighContrast};
      background-color: ${color.accent};
    `}

  ${(props) =>
    props.lighten &&
    css`
      color: ${color.font};
      background-color: ${color.background};
    `}
`
