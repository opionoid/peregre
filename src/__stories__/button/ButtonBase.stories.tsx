import { Story, Meta } from '@storybook/react'
import { ButtonBase, IButtonBaseProps } from 'src/components/actor/button/ButtonBase'
import MOCK_BUTTON_BASE_PROPS from 'src/__mocks__/components/actor/button/ButtonBase'
import { space } from 'src/assets/style'

export default {
  title: 'Components/actor/button/ButtonBase',
  component: ButtonBase,
  decorators: [(Story) => <div style={{ padding: '32px', backgroundColor: '#fef8e7' }}><Story/></div>]
} as Meta

const Template: Story<IButtonBaseProps> = (args) => <ButtonBase type={args.type} onClick={args.onClick}>{args.children}</ButtonBase>

export const Default = Template.bind({})
Default.args = {
  ...MOCK_BUTTON_BASE_PROPS,
  children: <p>ボタン</p>
}
export const HasChildren = Template.bind({})
HasChildren.args = {
  children: <div><h2>タイトル</h2><p style={{ marginTop: space.xxs }}>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p></div>
}
