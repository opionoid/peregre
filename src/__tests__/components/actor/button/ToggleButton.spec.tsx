import { ToggleButton } from 'src/components/actor/button/ToggleButton'
import MOCK_TOGGLE_BUTTON_PROPS from 'src/__mocks__/components/actor/button/ToggleButton'
import { render } from '@testing-library/react'

describe('actor/button', () => {
  const snapshot = render(<ToggleButton {...MOCK_TOGGLE_BUTTON_PROPS} />)

  it('default test', () => {
    expect(snapshot.baseElement).toMatchSnapshot()
  })
})
