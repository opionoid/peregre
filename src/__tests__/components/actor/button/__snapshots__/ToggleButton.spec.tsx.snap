// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actor/button default test 1`] = `
<body>
  <div>
    <div
      class="sc-gsDJrp kntnyI"
    >
      <button
        aria-disabled="false"
        class="sc-bdvvaa gazzRl"
        type="button"
      >
        <div
          class="sc-dkPtyc bVVxKk"
        >
          <p>
            ON
          </p>
          <img
            alt="でふぉ"
            class="sc-hKwCoD flvpqd"
            src="Hidden.svg"
          />
        </div>
      </button>
    </div>
  </div>
</body>
`;
